SorrTask64: Move Code Style doc to md

DanilYachmenev opened the pull request:
TasK #64 

Follow-up from https://github.com/sorrentum/sorrentum/issues/64#issuecomment-1519166797

Code style doc is wildly huge, so takes time to edit it

So far I could process 75% of the doc, until `5.14.6 Structure the code in terms of` section
Pls take a look and let's decide what else should be done here

gpsaggese commented on docs/Coding_Style_Guide.md:
```diff
@@ -0,0 +1,2166 @@
+
+[Always use imperative docstring and comments](#_h3504xf4mtjv)
+
+# <a name="_jriwruw1q2lb"></a><a name="_1oydvxmk6rm4"></a>**Meta**
```
Can you create a short Python script to convert

```
# <a name="_jriwruw1q2lb"></a><a name="_1oydvxmk6rm4"></a>**Meta**
```

into 

```
# Meta
```

DanilYachmenev replied:
Added a script simple script that removes TOH notations
Before it was giving Non-ASCII symbols but now seems all good
<img width="1076" alt="Снимок экрана 2023-05-02 в 22 26 03" src="https://user-images.githubusercontent.com/31514660/236518138-609a0d98-6310-4cc9-ae86-4f56837246c8.png">

My question is now: do we want to git rid of `**` type markers that make headers bold?
I've tried to implement it here and it looks worse IMO
Let me know if you want to get rid of them as well, then we need to discuss how to do it actually)

gpsaggese commented on docs/Coding_Style_Guide.md:
```diff
@@ -0,0 +1,2166 @@
+﻿# Sorrentum - Python Style Guide
+
+[Meta](#_1oydvxmk6rm4)
```
IMO you can remove the table of content generated by Google docs, since the linter should generate a better one automatically. Ok?

gpsaggese reviewed (commented):
Going in the right direction

DanilYachmenev commented on docs/Coding_Style_Guide.md:
```diff
@@ -0,0 +1,1965 @@
+# **High-Level Principles**
+In this paragraph we summarize the high-level principles that we follow for designing and implementing code and research. We should be careful in adding principles here. Ideally principles should be non-overlapping and generating all the other lower level principles we follow (like a basis for a vector space)
+
+- ### ***Follow the [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) principle***
+- ### ***Optimize for reading***
+  - Make code easy to read even if it is more difficult to write
+  - Code is written 1x and read 100x
+- ### ***Encapsulate what changes***
```
E.g. here principles are much more visible being bold

gpsaggese replied:
I agree, but we need to avoid tweaking the markdown style, otherwise everyone will start tweaking in different ways based on their preference. Markdown is rendered as it's rendered...

So let's remove the *** around the titles.

gpsaggese commented on dev_scripts/cleanup_scripts/SorrTask188_Move_Code_Style_doc_to_md.sh:
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash -xe
+
+replace_text.py \
```
Great idea re-using this script. I almost forgot about it.

gpsaggese reviewed (commented):
One tweak and merge. Good job!

DanilYachmenev commented:
- Updated README https://github.com/sorrentum/sorrentum/pull/227
- Updated Tecnical docs doc https://docs.google.com/document/d/14Ul5uWd7NU1zZ9ZLHp1TerLMN7MOfS5MS5bSX-2rBQ8/edit
- Original Gdoc is striked out https://docs.google.com/document/d/1R6jhFDbZKvyDdbnSQ9DA_N8417YF13hMI_Uw4quO4Xk/edit#heading=h.dfps6lwjmci4
